src/components/NumberField/NumberField.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Input, Label } from './styles';
+
+const NumberField = ({ label, value, onChange }) => {
+  const [inputValue, setInputValue] = useState(value);
```

Comment:
Is it works correctly? Don't you need `useEffect` where you will update state if a new value comes from parent component?